examples/example_prom/main.go

```diff
@@ -0,0 +1,221 @@
+
+$ go run main.go
+
+then curl the http address like this:
+
+$ curl localhost:8436/metrics
+# HELP snmp_about_info SNMP narrative metric with a default value of 1
+# TYPE snmp_about_info gauge
+snmp_about_info{contact="contact",site="hq",sysServices="1001110",target="10.12.0.1"} 1
+# HELP snmp_build_info A metric with a constant '1' value labeled by version, revision, branch, and goversion from which snmp was built.
+# TYPE snmp_build_info gauge
+snmp_build_info{branch="",goversion="go1.15.2",revision="",version=""} 1
+# HELP snmp_response_duration_seconds SNMP packet response latency
+# TYPE snmp_response_duration_seconds histogram
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.0005"} 0
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.001"} 9
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.0025"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.005"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.01"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.025"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.05"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.1"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.25"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="0.5"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="1"} 10
+snmp_response_duration_seconds_bucket{site="hq",target="10.12.0.1",le="+Inf"} 10
+snmp_response_duration_seconds_sum{site="hq",target="10.12.0.1"} 0.007858001
+snmp_response_duration_seconds_count{site="hq",target="10.12.0.1"} 10
+# HELP snmp_response_latency_seconds SNMP packet response latency
+# TYPE snmp_response_latency_seconds gauge
+snmp_response_latency_seconds{site="hq",target="10.12.0.1"} 0.000714456 1609641200001
+
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	g "github.com/gosnmp/gosnmp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/version"
+)
+
+func main() {
+	// Default is a pointer to a GoSNMP struct that contains sensible defaults
+	// eg port 161, community public, etc
+	g.Default.Target = "10.12.0.1"
+	err := g.Default.Connect()
+	if err != nil {
+		log.Fatalf("Connect() err: %v", err)
+	}
+	defer g.Default.Conn.Close()
+
+	// Function handles for collecting metrics on SNMP query latencies.
+	var sent time.Time
+	g.Default.OnSent = func(x *g.GoSNMP) {
+		sent = time.Now()
+	}
+	g.Default.OnRecv = func(x *g.GoSNMP) {
+		ObserveLatency(time.Since(sent).Seconds())
+	}
+
+	// Set constant prometheus labels
+	snmpLabels["target"] = g.Default.Target
+	snmpLabels["site"] = "hq"
```

Comment:
It's against best practices to include location labels in metrics. These are assigned in the Prometheus config.

Really, the best practice here is to not include any labels. It's a bit noisy for this simple example code and encourages users to do the same.